src/services/basic-primitive-on-map/basic-primitive-on-map.service.ts

```diff
@@ -0,0 +1,46 @@
+import { OnInit, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { SimpleDrawerService } from '../simple-drawer/simple-drawer.service';
+
+/**
+ *  Extend this class to create drawing on map components.
+ */
+export class BasicPrimitiveOnMap implements OnInit, OnChanges {
+	@Input()
+	props: any;
+
+	private selfPrimitive: any;
+
+	private selfPrimitiveIsDraw: boolean;
+
+	constructor(protected _drawer: SimpleDrawerService) {
+	}
+
+	ngOnInit(): void {
+		this.selfPrimitiveIsDraw = false;
+		this.drawOnMap();
+	}
+
+	ngOnChanges(changes: SimpleChanges) {
+		const props = changes['props'];
+		if (props.currentValue !== props.previousValue) {
+			this.updateOnMap();
+		}
+	}
+
+	drawOnMap() {
+		this.selfPrimitiveIsDraw =true;
+		return this.selfPrimitive = this._drawer.add(this.props);
+	}
+
+	removeFromMap() {
+		this.selfPrimitiveIsDraw = false;
+		return this._drawer.remove(this.selfPrimitive);
+	}
+
+	updateOnMap() {
+		if (this.selfPrimitiveIsDraw) {
+			this.removeFromMap();
```

Comment:
I can't use it cause its not fit to this kind of drawing.
Its better to remove and add.
It doesn't make a performance issues cause we talk about only few objects